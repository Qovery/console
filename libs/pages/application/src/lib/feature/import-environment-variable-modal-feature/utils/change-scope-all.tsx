import { APIVariableScopeEnum } from 'qovery-typescript-axios'
import { UseFormSetValue } from 'react-hook-form'

export function changeScopeForAll(
  value: APIVariableScopeEnum,
  setValue: UseFormSetValue<{ [key: string]: string }>,
  keys: (string | boolean)[]
): void {
  keys.forEach((key) => {
    setValue(key + '_scope', value?.toString())
  })
}
