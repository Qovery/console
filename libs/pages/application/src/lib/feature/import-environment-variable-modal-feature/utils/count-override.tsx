export function countOverride(data: { [key: string]: string }, existingEnvVarNames: (string | undefined)[]) {
  let count = 0
  Object.keys(data)
    .filter((key) => key.indexOf('_key') >= 0)
    .forEach((key) => {
      if (existingEnvVarNames.includes(data[key])) count++
    })

  return count
}
