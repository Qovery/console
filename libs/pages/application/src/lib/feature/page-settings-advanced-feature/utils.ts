import { ApplicationAdvancedSettings } from 'qovery-typescript-axios'
import { ApplicationEntity } from '@console/shared/interfaces'

export function initFormValues(keys: string[], application: ApplicationEntity): { [key: string]: string } {
  const values: { [key: string]: string } = {}
  keys.forEach((key) => {
    if (application.advanced_settings?.current_settings) {
      values[key] =
        application.advanced_settings.current_settings[key as keyof ApplicationAdvancedSettings]?.toString() || ''
    }
  })

  return values
}
