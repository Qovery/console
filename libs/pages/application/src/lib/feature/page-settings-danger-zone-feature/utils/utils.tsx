import { UseFormRegister, UseFormUnregister } from 'react-hook-form/dist/types/form'

export const onAddStorage = (
  register: UseFormRegister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  const id = Math.trunc(Math.random() * 10000).toString()

  register('size_' + id, { value: '' })
  register('type_' + id, { value: '' })
  register('path_' + id, { value: '' })

  return [...keys, id]
}

export const onRemove = (
  key: string,
  unregister: UseFormUnregister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  unregister('size_' + key)
  unregister('type_' + key)
  unregister('path_' + key)

  return keys.filter((k) => k !== key)
}
