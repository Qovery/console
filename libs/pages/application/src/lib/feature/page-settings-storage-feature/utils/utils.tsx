import { UseFormRegister, UseFormUnregister } from 'react-hook-form/dist/types/form'
import { ApplicationStorageStorage } from 'qovery-typescript-axios'

export const addStorage = (register: UseFormRegister<{ [key: string]: string | number }>, keys: string[]): string[] => {
  const id = Math.trunc(Math.random() * 10000).toString()

  register('size_' + id, { value: '' })
  register('type_' + id, { value: '' })
  register('path_' + id, { value: '' })

  return [...keys, id]
}

export const onRemove = (
  key: string,
  unregister: UseFormUnregister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  unregister('size_' + key)
  unregister('type_' + key)
  unregister('path_' + key)

  return keys.filter((k) => k !== key)
}

export const initStorage = (
  register: UseFormRegister<{ [key: string]: string | number }>,
  storages: ApplicationStorageStorage[]
): string[] => {
  const keys: string[] = []

  storages.forEach((storage) => {
    register('size_' + storage.id, { value: storage.size })
    register('type_' + storage.id, { value: storage.type })
    register('path_' + storage.id, { value: storage.mount_point })

    keys.push(storage.id || '')
  })

  return keys
}
