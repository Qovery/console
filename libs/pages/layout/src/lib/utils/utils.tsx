export function setCurrentOrganizationIdOnStorage(organizationId: string) {
  localStorage.setItem('currentOrganizationId', organizationId)
}

export function setCurrentProjectIdOnStorage(projectId: string) {
  localStorage.setItem('currentProjectId', projectId)
}
