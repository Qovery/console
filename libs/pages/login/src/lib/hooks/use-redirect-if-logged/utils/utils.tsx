export function getRedirectLoginUriFromStorage(): string | null {
  return localStorage.getItem('redirectLoginUri')
}

export function getCurrentOrganizationIdFromStorage(): string | null {
  return localStorage.getItem('currentOrganizationId')
}

export function getCurrentProjectIdFromStorage(): string | null {
  return localStorage.getItem('currentProjectId')
}
