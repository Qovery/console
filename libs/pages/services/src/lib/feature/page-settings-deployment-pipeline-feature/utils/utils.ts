import { type DeploymentStageResponse, type DeploymentStageServiceResponse } from 'qovery-typescript-axios'
import { type DraggableLocation } from 'react-beautiful-dnd'

export const reorderService = (
  stages: DeploymentStageResponse[],
  destinationIndex: number,
  startIndex: number,
  endIndex: number
): DeploymentStageResponse[] => {
  const cloneStages = [...stages]
  const currentStage = cloneStages[destinationIndex]
  const currentStageServices = [...(currentStage.services as DeploymentStageServiceResponse[])]

  const [removed] = currentStageServices.splice(startIndex, 1)
  currentStageServices.splice(endIndex, 0, removed)

  // add reorder on the clone stages array
  const result = cloneStages.map((stage, index) => {
    if (index === destinationIndex) {
      return { ...stage, services: currentStageServices }
    }
    return stage
  })

  return result
}

export const move = (
  stages: DeploymentStageResponse[],
  droppableSource: DraggableLocation,
  droppableDestination: DraggableLocation
): DeploymentStageResponse[] | undefined => {
  let result
  const cloneStages = [...stages]

  const sourceDroppableId = parseInt(droppableSource.droppableId, 10)
  const destinationDroppableId = parseInt(droppableDestination.droppableId, 10)

  // find the source stage
  const sourceStage = cloneStages.find((stage, index) => index === sourceDroppableId)

  if (sourceStage) {
    // find the destination stage
    const destinationStage = cloneStages.find((stage, index) => index === destinationDroppableId)

    if (destinationStage?.services && sourceStage?.services) {
      const cloneSourceServices = [...sourceStage.services]
      // remove the service from the source services
      const [serviceMoved] = cloneSourceServices.splice(droppableSource.index, 1)

      // add the service on the destination services on the droppable destination index
      const cloneDestinationServices = [...destinationStage.services]
      cloneDestinationServices.splice(droppableDestination.index, 0, serviceMoved)

      // add new services for source and destination on the stages
      result = cloneStages.map((stage, index) => {
        if (index === sourceDroppableId) {
          return { ...stage, services: cloneSourceServices }
        }
        if (index === destinationDroppableId) {
          return { ...stage, services: cloneDestinationServices }
        }
        return stage
      })
    }
  }

  return result
}
