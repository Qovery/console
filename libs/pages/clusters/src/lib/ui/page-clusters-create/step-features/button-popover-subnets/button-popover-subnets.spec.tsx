import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import ButtonPopoverSubnets from './button-popover-subnets'

describe('ButtonPopoverSubnets', () => {
  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(
      wrapWithReactHookForm(
        <ButtonPopoverSubnets
          sections={[
            {
              title: 'EKS subnets',
              name: 'eks_subnets',
            },
          ]}
        >
          EKS
        </ButtonPopoverSubnets>
      )
    )
    expect(baseElement).toBeTruthy()
  })

  it('should match snapshot', async () => {
    const { container, userEvent } = renderWithProviders(
      wrapWithReactHookForm(
        <ButtonPopoverSubnets
          sections={[
            {
              title: 'EKS subnets',
              name: 'eks_subnets',
            },
          ]}
        >
          EKS
        </ButtonPopoverSubnets>,
        {
          defaultValues: {
            eks_subnets: [{ A: 'test', B: 'test', C: 'test' }],
          },
        }
      ),
      {
        container: document.body,
      }
    )
    const button = screen.getByRole('button', { name: /eks/i })
    await userEvent.click(button)

    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with invalid state', async () => {
    const { container, userEvent } = renderWithProviders(
      wrapWithReactHookForm(
        <ButtonPopoverSubnets
          sections={[
            {
              title: 'EKS subnets',
              name: 'eks_subnets',
            },
          ]}
        >
          EKS
        </ButtonPopoverSubnets>,
        {
          defaultValues: {
            eks_subnets: [{ A: 'test', B: 'test', C: '' }],
          },
        }
      ),
      {
        container: document.body,
      }
    )
    const button = screen.getByRole('button', { name: /eks/i })
    await userEvent.click(button)

    expect(container).toMatchSnapshot()
  })
})
