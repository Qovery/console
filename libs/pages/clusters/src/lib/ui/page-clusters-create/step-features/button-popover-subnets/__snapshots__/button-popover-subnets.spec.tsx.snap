// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPopoverSubnets should match snapshot 1`] = `
<body>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <button
    aria-controls="radix-:rd:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded-full border dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 self-start border-green-500 bg-white"
    data-state="open"
    type="button"
  >
    EKS
  </button>
  <div
    data-radix-popper-content-wrapper=""
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 8px); min-width: max-content; --radix-popper-available-width: -20px; --radix-popper-available-height: -18px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
  >
    <div
      class="rounded bg-white shadow-xl outline-none data-[state=open]:data-[side=bottom]:animate-slidein-up-sm-faded data-[state=open]:data-[side=left]:animate-slidein-right-sm-faded data-[state=open]:data-[side=right]:animate-slidein-left-sm-faded data-[state=open]:data-[side=top]:animate-slidein-down-sm-faded dark:border dark:border-neutral-500 dark:bg-neutral-550 relative divide-y p-0 text-sm text-neutral-350 data-[state=open]:block data-[state=closed]:hidden"
      data-align="start"
      data-side="bottom"
      data-state="open"
      id="radix-:rd:"
      role="dialog"
      style="width: 648px; --radix-popover-content-transform-origin: var(--radix-popper-transform-origin); --radix-popover-content-available-width: var(--radix-popper-available-width); --radix-popover-content-available-height: var(--radix-popper-available-height); --radix-popover-trigger-width: var(--radix-popper-anchor-width); --radix-popover-trigger-height: var(--radix-popper-anchor-height);"
      tabindex="-1"
    >
      <div
        class="flex flex-col gap-3 p-4"
      >
        <h6
          class="font-medium text-neutral-400"
        >
          EKS subnets
        </h6>
        <ul
          class="flex flex-col gap-3"
        >
          <li
            class="grid grid-cols-[6fr_6fr_6fr_1fr] items-center gap-x-2 text-sm font-medium text-neutral-350"
          >
            <span>
              IDs zone A
            </span>
            <span>
              IDs zone B
            </span>
            <span>
              IDs zone C
            </span>
            <span />
          </li>
          <li
            class="grid grid-cols-[6fr_6fr_6fr_1fr] items-center gap-x-2"
          >
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.A"
                  type="text"
                  value="test"
                />
              </div>
            </div>
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.B"
                  type="text"
                  value="test"
                />
              </div>
            </div>
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.C"
                  type="text"
                  value="test"
                />
              </div>
            </div>
            <button
              class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none w-full"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fa-regular fa-trash-can text-base"
              />
            </button>
          </li>
        </ul>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none self-end"
          type="button"
        >
          Add subnets
          <i
            aria-hidden="true"
            class="fa-solid fa-plus ml-2 text-base"
          />
        </button>
      </div>
      <div
        class="flex flex-col gap-4 p-4 text-right text-base"
      >
        <div>
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none mr-1"
            type="button"
          >
            Cancel
          </button>
          <span
            type="button"
          >
            <button
              class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-9 px-3 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
              type="button"
            >
              Confirm
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`ButtonPopoverSubnets should match snapshot with invalid state 1`] = `
<body>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <button
    aria-controls="radix-:rk:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded-full border dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 self-start border-red-500 bg-white"
    data-state="open"
    type="button"
  >
    EKS
  </button>
  <div
    data-radix-popper-content-wrapper=""
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 8px); min-width: max-content; --radix-popper-available-width: -20px; --radix-popper-available-height: -18px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
  >
    <div
      class="rounded bg-white shadow-xl outline-none data-[state=open]:data-[side=bottom]:animate-slidein-up-sm-faded data-[state=open]:data-[side=left]:animate-slidein-right-sm-faded data-[state=open]:data-[side=right]:animate-slidein-left-sm-faded data-[state=open]:data-[side=top]:animate-slidein-down-sm-faded dark:border dark:border-neutral-500 dark:bg-neutral-550 relative divide-y p-0 text-sm text-neutral-350 data-[state=open]:block data-[state=closed]:hidden"
      data-align="start"
      data-side="bottom"
      data-state="open"
      id="radix-:rk:"
      role="dialog"
      style="width: 648px; --radix-popover-content-transform-origin: var(--radix-popper-transform-origin); --radix-popover-content-available-width: var(--radix-popper-available-width); --radix-popover-content-available-height: var(--radix-popper-available-height); --radix-popover-trigger-width: var(--radix-popper-anchor-width); --radix-popover-trigger-height: var(--radix-popper-anchor-height);"
      tabindex="-1"
    >
      <div
        class="flex flex-col gap-3 p-4"
      >
        <h6
          class="font-medium text-neutral-400"
        >
          EKS subnets
        </h6>
        <ul
          class="flex flex-col gap-3"
        >
          <li
            class="grid grid-cols-[6fr_6fr_6fr_1fr] items-center gap-x-2 text-sm font-medium text-neutral-350"
          >
            <span>
              IDs zone A
            </span>
            <span>
              IDs zone B
            </span>
            <span>
              IDs zone C
            </span>
            <span />
          </li>
          <li
            class="grid grid-cols-[6fr_6fr_6fr_1fr] items-center gap-x-2"
          >
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.A"
                  type="text"
                  value="test"
                />
              </div>
            </div>
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.B"
                  type="text"
                  value="test"
                />
              </div>
            </div>
            <div
              class="w-full"
              data-testid="input-small-wrapper"
            >
              <div
                class="input input--small flex-grow"
                data-testid="input"
              >
                <label
                  class="hidden"
                />
                <input
                  class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                  data-testid="input-value"
                  name="eks_subnets.0.C"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none w-full"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fa-regular fa-trash-can text-base"
              />
            </button>
          </li>
        </ul>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none self-end"
          type="button"
        >
          Add subnets
          <i
            aria-hidden="true"
            class="fa-solid fa-plus ml-2 text-base"
          />
        </button>
      </div>
      <div
        class="flex flex-col gap-4 p-4 text-right text-base"
      >
        <div>
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none mr-1"
            type="button"
          >
            Cancel
          </button>
          <span
            type="button"
          >
            <button
              class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-9 px-3 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
              data-state="closed"
              disabled=""
              type="button"
            >
              Confirm
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
